spec/ulid_spec.cr

```diff
@@ -57,4 +57,82 @@ describe ULID do
       end
     end
   end
+
+  describe ".valid?" do
+    it "should validate a valid string" do
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should be case insensitive" do
+      ULID.valid?("01b3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should detect incorrect length" do
+      ULID.valid?("0").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ32").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ").should be_false
+      ULID.valid?("!@#$%^&*(").should be_false
+      ULID.valid?("abcde").should be_false
+      ULID.valid?("1234567890").should be_false
+      ULID.valid?("").should be_false
+    end
+
+    it "should detect invalid characters" do
+      ULID.valid?("01!3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01I3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01O3EAF48P97R8MP9WS8MHDTZ3").should be_false
+    end
+  end
+
+  describe ".validate!" do
+    it "should validate a valid string" do
+      ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should be case insensitive" do
+      ULID.validate!("01b3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should detect incorrect length" do
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("0") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ32") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("!@#$%^&*(") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("abcde") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("1234567890") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("") }
```

Comment:
just imo might be easier to write these specs with loop, i.e.

["0", "abcde", ...].each { |i|
 expect_raises(ULID::IncorrectLength) {ULID.validate!(i)}
}

Reply:
nice